File: GitHubSearch/GitHubRepository.swift
//
//  GitHubRepository.swift
//  GitHubSearch
//
//  Created by horimislime on 2018/01/13.
//  Copyright © 2018 horimislime. All rights reserved.
//

import Foundation

struct SearchRepositoryResponse: Decodable {
    
    var items: [RepositoryResponse]
    
    enum CodingKeys: String, CodingKey {
        case items
    }
}

struct RepositoryResponse: Decodable {
    
    var fullName: String
    var description: String?
    var language: String?
    var starCount: Int
    var updatedAt: Date
    var url: URL
    
    enum CodingKeys: String, CodingKey {
        case fullName = "full_name"
        case description
        case language
        case starCount = "stargazers_count"
        case updatedAt = "updated_at"
        case url = "html_url"
    }
}

File: GitHubSearchFlutter/lib/model/github_repository.dart
class GitHubRepository {
  final String name;
  final String description;
  final String language;
  final int starCount;
  final DateTime updatedAt;
  final String url;

  const GitHubRepository(
      {this.name,
      this.description,
      this.language,
      this.starCount,
      this.updatedAt,
      this.url});

  factory GitHubRepository.fromJson(Map<String, dynamic> json) {
    return GitHubRepository(
      name: json['name'],
      description: json['description'],
      language: json['language'],
      starCount: json['starCount'],
      updatedAt: json['updatedAt'],
      url: json['url'],
    );
  }
}
